scripts/common_windows.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# 	http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+set -uo pipefail
+
+#############################################################################################################
+# Helper methods to set the required variables.
+#############################################################################################################
+# Constant
+# Number of presently supported OS versions.
+SUPPORTED_WINDOWS_OS_RELEASES_NUMBER=2
+
+# Sets ALL_AWS_FOR_FLUENT_BIT_VERSIONS variable with all the supported fluent-bit versions.
+set_all_supported_versions() {
+  ALL_AWS_FOR_FLUENT_BIT_VERSIONS=$(cat < ../windows.versions | jq -c -r '.windows[].version')
+}
+
+# Sets AWS_FOR_FLUENT_BIT_STABLE_VERSION variable with the stable version.
+set_stable_version() {
+  # Find the stable version.
+  AWS_FOR_FLUENT_BIT_STABLE_VERSION=$(cat ../AWS_FOR_FLUENT_BIT_STABLE_VERSION)
+}
+
+# Sets AWS_FOR_FLUENT_BIT_LATEST_VERSION variable with the latest version.
+set_latest_version() {
+  # Find the latest version.
+  AWS_FOR_FLUENT_BIT_LATEST_VERSION=""
+  while read -r version; do
+    if [[ "$AWS_FOR_FLUENT_BIT_LATEST_VERSION" < "$version" ]]; then
+      AWS_FOR_FLUENT_BIT_LATEST_VERSION=$version
+    fi
+  done <<< "$(echo "$ALL_AWS_FOR_FLUENT_BIT_VERSIONS")"
+}
+
+# Sets AWS_FOR_FLUENT_BIT_VERSION_DOCKERHUB variable with the latest version available in dockerhub.
+set_dockerhub_version() {
+  # Find the dockerhub version.
+  docker_hub_image_tags=$(curl -s -S 'https://registry.hub.docker.com/v2/repositories/amazon/aws-for-fluent-bit/tags/?page=1&page_size=250' | jq -r '.results[].name')
```

Comment:
When we push to dockerhub are they full layers or are they foreign layers?

Reply:
So as discussed offline, currently we are pushing full layers to dockerhub.